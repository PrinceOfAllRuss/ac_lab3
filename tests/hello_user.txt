.data:
    start "< What is your name?$> "
    end "< Hello, "
.code:
    mov start
    loop_1:
        jz 4
        out 1 str
        jmp loop_1
    in 0 55
    loop_2:
        jz 8
        out 1 str
        jmp loop_2
    mov end
    loop_3:
        jz 12
        out 1 str
        jmp loop_3
    mov 55
    loop_4:
        jz break
        out 1 str
        jmp loop_4
    break:
        halt