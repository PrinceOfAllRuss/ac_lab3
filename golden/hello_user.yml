in_source: |-
  .data:
    start "< What is your name?$> "
    end "< Hello, "
  .code:
      mov start
      loop_1:
          jz 4
          out 1 str
          jmp loop_1
      in 0 55
      loop_2:
          jz 8
          out 1 str
          jmp loop_2
      mov end
      loop_3:
          jz 12
          out 1 str
          jmp loop_3
      mov 55
      loop_4:
          jz break
          out 1 str
          jmp loop_4
      break:
          halt
in_stdin: |-
  Fedor
out_code: |-
  [{"index": 0, "operation": "mov", "arg": [17]},
  {"index": 1, "operation": "jz", "arg": [4]},
  {"index": 2, "operation": "out", "arg": [1, "str"]},
  {"index": 3, "operation": "jmp", "arg": [1]},
  {"index": 4, "operation": "in", "arg": [0, 55]},
  {"index": 5, "operation": "jz", "arg": [8]},
  {"index": 6, "operation": "out", "arg": [1, "str"]},
  {"index": 7, "operation": "jmp", "arg": [5]},
  {"index": 8, "operation": "mov", "arg": [41]},
  {"index": 9, "operation": "jz", "arg": [12]},
  {"index": 10, "operation": "out", "arg": [1, "str"]},
  {"index": 11, "operation": "jmp", "arg": [9]},
  {"index": 12, "operation": "mov", "arg": [55]},
  {"index": 13, "operation": "jz", "arg": [16]},
  {"index": 14, "operation": "out", "arg": [1, "str"]},
  {"index": 15, "operation": "jmp", "arg": [13]},
  {"index": 16, "operation": "halt"},
  {"index": 17, "data": 60},
  {"index": 18, "data": 32},
  {"index": 19, "data": 87},
  {"index": 20, "data": 104},
  {"index": 21, "data": 97},
  {"index": 22, "data": 116},
  {"index": 23, "data": 32},
  {"index": 24, "data": 105},
  {"index": 25, "data": 115},
  {"index": 26, "data": 32},
  {"index": 27, "data": 121},
  {"index": 28, "data": 111},
  {"index": 29, "data": 117},
  {"index": 30, "data": 114},
  {"index": 31, "data": 32},
  {"index": 32, "data": 110},
  {"index": 33, "data": 97},
  {"index": 34, "data": 109},
  {"index": 35, "data": 101},
  {"index": 36, "data": 63},
  {"index": 37, "data": 10},
  {"index": 38, "data": 62},
  {"index": 39, "data": 32},
  {"index": 41, "data": 0},
  {"index": 41, "data": 60},
  {"index": 42, "data": 32},
  {"index": 43, "data": 72},
  {"index": 44, "data": 101},
  {"index": 45, "data": 108},
  {"index": 46, "data": 108},
  {"index": 47, "data": 111},
  {"index": 48, "data": 44},
  {"index": 49, "data": 32},
  {"index": 51, "data": 0}]
out_stdout: |+
  source LoC: 26 code instr: 16
  ============================================================
  < What is your name?
  > Fedor
  < Hello, Fedor

out_log: |
  DEBUG   control_unit:start         PC: 0 TICK: 0 P_ADDR: 0 MEM_ADDR: 0 ACC: 0 COMMAND: mov [17]
  DEBUG   control_unit:start         PC: 1 TICK: 2 P_ADDR: 1 MEM_ADDR: 17 ACC: 0 COMMAND: jz [4]
  DEBUG   control_unit:start         PC: 2 TICK: 3 P_ADDR: 2 MEM_ADDR: 17 ACC: 0 COMMAND: out [1, 'str']
  DEBUG   isa:perform       out: "" << "<"
  DEBUG   control_unit:start         PC: 3 TICK: 7 P_ADDR: 3 MEM_ADDR: 18 ACC: 60 COMMAND: jmp [1]
  DEBUG   control_unit:start         PC: 4 TICK: 8 P_ADDR: 1 MEM_ADDR: 18 ACC: 60 COMMAND: jz [4]
  DEBUG   control_unit:start         PC: 5 TICK: 9 P_ADDR: 2 MEM_ADDR: 18 ACC: 60 COMMAND: out [1, 'str']
  DEBUG   isa:perform       out: "<" << " "
  DEBUG   control_unit:start         PC: 6 TICK: 13 P_ADDR: 3 MEM_ADDR: 19 ACC: 32 COMMAND: jmp [1]
  DEBUG   control_unit:start         PC: 7 TICK: 14 P_ADDR: 1 MEM_ADDR: 19 ACC: 32 COMMAND: jz [4]
  DEBUG   control_unit:start         PC: 8 TICK: 15 P_ADDR: 2 MEM_ADDR: 19 ACC: 32 COMMAND: out [1, 'str']
  DEBUG   isa:perform       out: "< " << "W"
  DEBUG   control_unit:start         PC: 9 TICK: 19 P_ADDR: 3 MEM_ADDR: 20 ACC: 87 COMMAND: jmp [1]
  DEBUG   control_unit:start         PC: 10 TICK: 20 P_ADDR: 1 MEM_ADDR: 20 ACC: 87 COMMAND: jz [4]
  DEBUG   control_unit:start         PC: 11 TICK: 21 P_ADDR: 2 MEM_ADDR: 20 ACC: 87 COMMAND: out [1, 'str']
  DEBUG   isa:perform       out: "< W" << "h"
  DEBUG   control_unit:start         PC: 12 TICK: 25 P_ADDR: 3 MEM_ADDR: 21 ACC: 104 COMMAND: jmp [1]
  DEBUG   control_unit:start         PC: 13 TICK: 26 P_ADDR: 1 MEM_ADDR: 21 ACC: 104 COMMAND: jz [4]
  DEBUG   control_unit:start         PC: 14 TICK: 27 P_ADDR: 2 MEM_ADDR: 21 ACC: 104 COMMAND: out [1, 'str']
  DEBUG   isa:perform       out: "< Wh" << "a"
  DEBUG   control_unit:start         PC: 15 TICK: 31 P_ADDR: 3 MEM_ADDR: 22 ACC: 97 COMMAND: jmp [1]
  DEBUG   control_unit:start         PC: 16 TICK: 32 P_ADDR: 1 MEM_ADDR: 22 ACC: 97 COMMAND: jz [4]
  DEBUG   control_unit:start         PC: 17 TICK: 33 P_ADDR: 2 MEM_ADDR: 22 ACC: 97 COMMAND: out [1, 'str']
  DEBUG   isa:perform       out: "< Wha" << "t"
  DEBUG   control_unit:start         PC: 18 TICK: 37 P_ADDR: 3 MEM_ADDR: 23 ACC: 116 COMMAND: jmp [1]
  DEBUG   control_unit:start         PC: 19 TICK: 38 P_ADDR: 1 MEM_ADDR: 23 ACC: 116 COMMAND: jz [4]
  DEBUG   control_unit:start         PC: 20 TICK: 39 P_ADDR: 2 MEM_ADDR: 23 ACC: 116 COMMAND: out [1, 'str']
  DEBUG   isa:perform       out: "< What" << " "
  DEBUG   control_unit:start         PC: 21 TICK: 43 P_ADDR: 3 MEM_ADDR: 24 ACC: 32 COMMAND: jmp [1]
  DEBUG   control_unit:start         PC: 22 TICK: 44 P_ADDR: 1 MEM_ADDR: 24 ACC: 32 COMMAND: jz [4]
  DEBUG   control_unit:start         PC: 23 TICK: 45 P_ADDR: 2 MEM_ADDR: 24 ACC: 32 COMMAND: out [1, 'str']
  DEBUG   isa:perform       out: "< What " << "i"
  DEBUG   control_unit:start         PC: 24 TICK: 49 P_ADDR: 3 MEM_ADDR: 25 ACC: 105 COMMAND: jmp [1]
  DEBUG   control_unit:start         PC: 25 TICK: 50 P_ADDR: 1 MEM_ADDR: 25 ACC: 105 COMMAND: jz [4]
  DEBUG   control_unit:start         PC: 26 TICK: 51 P_ADDR: 2 MEM_ADDR: 25 ACC: 105 COMMAND: out [1, 'str']
  DEBUG   isa:perform       out: "< What i" << "s"
  DEBUG   control_unit:start         PC: 27 TICK: 55 P_ADDR: 3 MEM_ADDR: 26 ACC: 115 COMMAND: jmp [1]
  DEBUG   control_unit:start         PC: 28 TICK: 56 P_ADDR: 1 MEM_ADDR: 26 ACC: 115 COMMAND: jz [4]
  DEBUG   control_unit:start         PC: 29 TICK: 57 P_ADDR: 2 MEM_ADDR: 26 ACC: 115 COMMAND: out [1, 'str']
  DEBUG   isa:perform       out: "< What is" << " "
  DEBUG   control_unit:start         PC: 30 TICK: 61 P_ADDR: 3 MEM_ADDR: 27 ACC: 32 COMMAND: jmp [1]
  DEBUG   control_unit:start         PC: 31 TICK: 62 P_ADDR: 1 MEM_ADDR: 27 ACC: 32 COMMAND: jz [4]
  DEBUG   control_unit:start         PC: 32 TICK: 63 P_ADDR: 2 MEM_ADDR: 27 ACC: 32 COMMAND: out [1, 'str']
  DEBUG   isa:perform       out: "< What is " << "y"
  DEBUG   control_unit:start         PC: 33 TICK: 67 P_ADDR: 3 MEM_ADDR: 28 ACC: 121 COMMAND: jmp [1]
  DEBUG   control_unit:start         PC: 34 TICK: 68 P_ADDR: 1 MEM_ADDR: 28 ACC: 121 COMMAND: jz [4]
  DEBUG   control_unit:start         PC: 35 TICK: 69 P_ADDR: 2 MEM_ADDR: 28 ACC: 121 COMMAND: out [1, 'str']
  DEBUG   isa:perform       out: "< What is y" << "o"
  DEBUG   control_unit:start         PC: 36 TICK: 73 P_ADDR: 3 MEM_ADDR: 29 ACC: 111 COMMAND: jmp [1]
  DEBUG   control_unit:start         PC: 37 TICK: 74 P_ADDR: 1 MEM_ADDR: 29 ACC: 111 COMMAND: jz [4]
  DEBUG   control_unit:start         PC: 38 TICK: 75 P_ADDR: 2 MEM_ADDR: 29 ACC: 111 COMMAND: out [1, 'str']
  DEBUG   isa:perform       out: "< What is yo" << "u"
  DEBUG   control_unit:start         PC: 39 TICK: 79 P_ADDR: 3 MEM_ADDR: 30 ACC: 117 COMMAND: jmp [1]
  DEBUG   control_unit:start         PC: 40 TICK: 80 P_ADDR: 1 MEM_ADDR: 30 ACC: 117 COMMAND: jz [4]
  DEBUG   control_unit:start         PC: 41 TICK: 81 P_ADDR: 2 MEM_ADDR: 30 ACC: 117 COMMAND: out [1, 'str']
  DEBUG   isa:perform       out: "< What is you" << "r"
  DEBUG   control_unit:start         PC: 42 TICK: 85 P_ADDR: 3 MEM_ADDR: 31 ACC: 114 COMMAND: jmp [1]
  DEBUG   control_unit:start         PC: 43 TICK: 86 P_ADDR: 1 MEM_ADDR: 31 ACC: 114 COMMAND: jz [4]
  DEBUG   control_unit:start         PC: 44 TICK: 87 P_ADDR: 2 MEM_ADDR: 31 ACC: 114 COMMAND: out [1, 'str']
  DEBUG   isa:perform       out: "< What is your" << " "
  DEBUG   control_unit:start         PC: 45 TICK: 91 P_ADDR: 3 MEM_ADDR: 32 ACC: 32 COMMAND: jmp [1]
  DEBUG   control_unit:start         PC: 46 TICK: 92 P_ADDR: 1 MEM_ADDR: 32 ACC: 32 COMMAND: jz [4]
  DEBUG   control_unit:start         PC: 47 TICK: 93 P_ADDR: 2 MEM_ADDR: 32 ACC: 32 COMMAND: out [1, 'str']
  DEBUG   isa:perform       out: "< What is your " << "n"
  DEBUG   control_unit:start         PC: 48 TICK: 97 P_ADDR: 3 MEM_ADDR: 33 ACC: 110 COMMAND: jmp [1]
  DEBUG   control_unit:start         PC: 49 TICK: 98 P_ADDR: 1 MEM_ADDR: 33 ACC: 110 COMMAND: jz [4]
  DEBUG   control_unit:start         PC: 50 TICK: 99 P_ADDR: 2 MEM_ADDR: 33 ACC: 110 COMMAND: out [1, 'str']
  DEBUG   isa:perform       out: "< What is your n" << "a"
  DEBUG   control_unit:start         PC: 51 TICK: 103 P_ADDR: 3 MEM_ADDR: 34 ACC: 97 COMMAND: jmp [1]
  DEBUG   control_unit:start         PC: 52 TICK: 104 P_ADDR: 1 MEM_ADDR: 34 ACC: 97 COMMAND: jz [4]
  DEBUG   control_unit:start         PC: 53 TICK: 105 P_ADDR: 2 MEM_ADDR: 34 ACC: 97 COMMAND: out [1, 'str']
  DEBUG   isa:perform       out: "< What is your na" << "m"
  DEBUG   control_unit:start         PC: 54 TICK: 109 P_ADDR: 3 MEM_ADDR: 35 ACC: 109 COMMAND: jmp [1]
  DEBUG   control_unit:start         PC: 55 TICK: 110 P_ADDR: 1 MEM_ADDR: 35 ACC: 109 COMMAND: jz [4]
  DEBUG   control_unit:start         PC: 56 TICK: 111 P_ADDR: 2 MEM_ADDR: 35 ACC: 109 COMMAND: out [1, 'str']
  DEBUG   isa:perform       out: "< What is your nam" << "e"
  DEBUG   control_unit:start         PC: 57 TICK: 115 P_ADDR: 3 MEM_ADDR: 36 ACC: 101 COMMAND: jmp [1]
  DEBUG   control_unit:start         PC: 58 TICK: 116 P_ADDR: 1 MEM_ADDR: 36 ACC: 101 COMMAND: jz [4]
  DEBUG   control_unit:start         PC: 59 TICK: 117 P_ADDR: 2 MEM_ADDR: 36 ACC: 101 COMMAND: out [1, 'str']
  DEBUG   isa:perform       out: "< What is your name" << "?"
  DEBUG   control_unit:start         PC: 60 TICK: 121 P_ADDR: 3 MEM_ADDR: 37 ACC: 63 COMMAND: jmp [1]
  DEBUG   control_unit:start         PC: 61 TICK: 122 P_ADDR: 1 MEM_ADDR: 37 ACC: 63 COMMAND: jz [4]
  DEBUG   control_unit:start         PC: 62 TICK: 123 P_ADDR: 2 MEM_ADDR: 37 ACC: 63 COMMAND: out [1, 'str']
  DEBUG   isa:perform       out: "< What is your name?" << "\n"
  DEBUG   control_unit:start         PC: 63 TICK: 127 P_ADDR: 3 MEM_ADDR: 38 ACC: 10 COMMAND: jmp [1]
  DEBUG   control_unit:start         PC: 64 TICK: 128 P_ADDR: 1 MEM_ADDR: 38 ACC: 10 COMMAND: jz [4]
  DEBUG   control_unit:start         PC: 65 TICK: 129 P_ADDR: 2 MEM_ADDR: 38 ACC: 10 COMMAND: out [1, 'str']
  DEBUG   isa:perform       out: "< What is your name?\n" << ">"
  DEBUG   control_unit:start         PC: 66 TICK: 133 P_ADDR: 3 MEM_ADDR: 39 ACC: 62 COMMAND: jmp [1]
  DEBUG   control_unit:start         PC: 67 TICK: 134 P_ADDR: 1 MEM_ADDR: 39 ACC: 62 COMMAND: jz [4]
  DEBUG   control_unit:start         PC: 68 TICK: 135 P_ADDR: 2 MEM_ADDR: 39 ACC: 62 COMMAND: out [1, 'str']
  DEBUG   isa:perform       out: "< What is your name?\n>" << " "
  DEBUG   control_unit:start         PC: 69 TICK: 139 P_ADDR: 3 MEM_ADDR: 40 ACC: 32 COMMAND: jmp [1]
  DEBUG   control_unit:start         PC: 70 TICK: 140 P_ADDR: 1 MEM_ADDR: 40 ACC: 32 COMMAND: jz [4]
  DEBUG   control_unit:start         PC: 71 TICK: 141 P_ADDR: 2 MEM_ADDR: 40 ACC: 32 COMMAND: out [1, 'str']
  DEBUG   control_unit:start         PC: 72 TICK: 143 P_ADDR: 3 MEM_ADDR: 40 ACC: 0 COMMAND: jmp [1]
  DEBUG   control_unit:start         PC: 73 TICK: 144 P_ADDR: 1 MEM_ADDR: 40 ACC: 0 COMMAND: jz [4]
  DEBUG   control_unit:start         PC: 74 TICK: 146 P_ADDR: 4 MEM_ADDR: 40 ACC: 0 COMMAND: in [0, 55]
  DEBUG   isa:perform       input: "F"
  DEBUG   isa:perform       input: "e"
  DEBUG   isa:perform       input: "d"
  DEBUG   isa:perform       input: "o"
  DEBUG   isa:perform       input: "r"
  DEBUG   isa:perform       input: "\n"
  DEBUG   control_unit:start         PC: 75 TICK: 162 P_ADDR: 5 MEM_ADDR: 55 ACC: 0 COMMAND: jz [8]
  DEBUG   control_unit:start         PC: 76 TICK: 163 P_ADDR: 6 MEM_ADDR: 55 ACC: 0 COMMAND: out [1, 'str']
  DEBUG   isa:perform       out: "< What is your name?\n> " << "F"
  DEBUG   control_unit:start         PC: 77 TICK: 167 P_ADDR: 7 MEM_ADDR: 56 ACC: 70 COMMAND: jmp [5]
  DEBUG   control_unit:start         PC: 78 TICK: 168 P_ADDR: 5 MEM_ADDR: 56 ACC: 70 COMMAND: jz [8]
  DEBUG   control_unit:start         PC: 79 TICK: 169 P_ADDR: 6 MEM_ADDR: 56 ACC: 70 COMMAND: out [1, 'str']
  DEBUG   isa:perform       out: "< What is your name?\n> F" << "e"
  DEBUG   control_unit:start         PC: 80 TICK: 173 P_ADDR: 7 MEM_ADDR: 57 ACC: 101 COMMAND: jmp [5]
  DEBUG   control_unit:start         PC: 81 TICK: 174 P_ADDR: 5 MEM_ADDR: 57 ACC: 101 COMMAND: jz [8]
  DEBUG   control_unit:start         PC: 82 TICK: 175 P_ADDR: 6 MEM_ADDR: 57 ACC: 101 COMMAND: out [1, 'str']
  DEBUG   isa:perform       out: "< What is your name?\n> Fe" << "d"
  DEBUG   control_unit:start         PC: 83 TICK: 179 P_ADDR: 7 MEM_ADDR: 58 ACC: 100 COMMAND: jmp [5]
  DEBUG   control_unit:start         PC: 84 TICK: 180 P_ADDR: 5 MEM_ADDR: 58 ACC: 100 COMMAND: jz [8]
  DEBUG   control_unit:start         PC: 85 TICK: 181 P_ADDR: 6 MEM_ADDR: 58 ACC: 100 COMMAND: out [1, 'str']
  DEBUG   isa:perform       out: "< What is your name?\n> Fed" << "o"
  DEBUG   control_unit:start         PC: 86 TICK: 185 P_ADDR: 7 MEM_ADDR: 59 ACC: 111 COMMAND: jmp [5]
  DEBUG   control_unit:start         PC: 87 TICK: 186 P_ADDR: 5 MEM_ADDR: 59 ACC: 111 COMMAND: jz [8]
  DEBUG   control_unit:start         PC: 88 TICK: 187 P_ADDR: 6 MEM_ADDR: 59 ACC: 111 COMMAND: out [1, 'str']
  DEBUG   isa:perform       out: "< What is your name?\n> Fedo" << "r"
  DEBUG   control_unit:start         PC: 89 TICK: 191 P_ADDR: 7 MEM_ADDR: 60 ACC: 114 COMMAND: jmp [5]
  DEBUG   control_unit:start         PC: 90 TICK: 192 P_ADDR: 5 MEM_ADDR: 60 ACC: 114 COMMAND: jz [8]
  DEBUG   control_unit:start         PC: 91 TICK: 193 P_ADDR: 6 MEM_ADDR: 60 ACC: 114 COMMAND: out [1, 'str']
  DEBUG   isa:perform       out: "< What is your name?\n> Fedor" << "\n"
  DEBUG   control_unit:start         PC: 92 TICK: 197 P_ADDR: 7 MEM_ADDR: 61 ACC: 10 COMMAND: jmp [5]
  DEBUG   control_unit:start         PC: 93 TICK: 198 P_ADDR: 5 MEM_ADDR: 61 ACC: 10 COMMAND: jz [8]
  DEBUG   control_unit:start         PC: 94 TICK: 199 P_ADDR: 6 MEM_ADDR: 61 ACC: 10 COMMAND: out [1, 'str']
  DEBUG   control_unit:start         PC: 95 TICK: 201 P_ADDR: 7 MEM_ADDR: 61 ACC: 0 COMMAND: jmp [5]
  DEBUG   control_unit:start         PC: 96 TICK: 202 P_ADDR: 5 MEM_ADDR: 61 ACC: 0 COMMAND: jz [8]
  DEBUG   control_unit:start         PC: 97 TICK: 204 P_ADDR: 8 MEM_ADDR: 61 ACC: 0 COMMAND: mov [41]
  DEBUG   control_unit:start         PC: 98 TICK: 206 P_ADDR: 9 MEM_ADDR: 41 ACC: 0 COMMAND: jz [12]
  DEBUG   control_unit:start         PC: 99 TICK: 207 P_ADDR: 10 MEM_ADDR: 41 ACC: 0 COMMAND: out [1, 'str']
  DEBUG   isa:perform       out: "< What is your name?\n> Fedor\n" << "<"
  DEBUG   control_unit:start         PC: 100 TICK: 211 P_ADDR: 11 MEM_ADDR: 42 ACC: 60 COMMAND: jmp [9]
  DEBUG   control_unit:start         PC: 101 TICK: 212 P_ADDR: 9 MEM_ADDR: 42 ACC: 60 COMMAND: jz [12]
  DEBUG   control_unit:start         PC: 102 TICK: 213 P_ADDR: 10 MEM_ADDR: 42 ACC: 60 COMMAND: out [1, 'str']
  DEBUG   isa:perform       out: "< What is your name?\n> Fedor\n<" << " "
  DEBUG   control_unit:start         PC: 103 TICK: 217 P_ADDR: 11 MEM_ADDR: 43 ACC: 32 COMMAND: jmp [9]
  DEBUG   control_unit:start         PC: 104 TICK: 218 P_ADDR: 9 MEM_ADDR: 43 ACC: 32 COMMAND: jz [12]
  DEBUG   control_unit:start         PC: 105 TICK: 219 P_ADDR: 10 MEM_ADDR: 43 ACC: 32 COMMAND: out [1, 'str']
  DEBUG   isa:perform       out: "< What is your name?\n> Fedor\n< " << "H"
  DEBUG   control_unit:start         PC: 106 TICK: 223 P_ADDR: 11 MEM_ADDR: 44 ACC: 72 COMMAND: jmp [9]
  DEBUG   control_unit:start         PC: 107 TICK: 224 P_ADDR: 9 MEM_ADDR: 44 ACC: 72 COMMAND: jz [12]
  DEBUG   control_unit:start         PC: 108 TICK: 225 P_ADDR: 10 MEM_ADDR: 44 ACC: 72 COMMAND: out [1, 'str']
  DEBUG   isa:perform       out: "< What is your name?\n> Fedor\n< H" << "e"
  DEBUG   control_unit:start         PC: 109 TICK: 229 P_ADDR: 11 MEM_ADDR: 45 ACC: 101 COMMAND: jmp [9]
  DEBUG   control_unit:start         PC: 110 TICK: 230 P_ADDR: 9 MEM_ADDR: 45 ACC: 101 COMMAND: jz [12]
  DEBUG   control_unit:start         PC: 111 TICK: 231 P_ADDR: 10 MEM_ADDR: 45 ACC: 101 COMMAND: out [1, 'str']
  DEBUG   isa:perform       out: "< What is your name?\n> Fedor\n< He" << "l"
  DEBUG   control_unit:start         PC: 112 TICK: 235 P_ADDR: 11 MEM_ADDR: 46 ACC: 108 COMMAND: jmp [9]
  DEBUG   control_unit:start         PC: 113 TICK: 236 P_ADDR: 9 MEM_ADDR: 46 ACC: 108 COMMAND: jz [12]
  DEBUG   control_unit:start         PC: 114 TICK: 237 P_ADDR: 10 MEM_ADDR: 46 ACC: 108 COMMAND: out [1, 'str']
  DEBUG   isa:perform       out: "< What is your name?\n> Fedor\n< Hel" << "l"
  DEBUG   control_unit:start         PC: 115 TICK: 241 P_ADDR: 11 MEM_ADDR: 47 ACC: 108 COMMAND: jmp [9]
  DEBUG   control_unit:start         PC: 116 TICK: 242 P_ADDR: 9 MEM_ADDR: 47 ACC: 108 COMMAND: jz [12]
  DEBUG   control_unit:start         PC: 117 TICK: 243 P_ADDR: 10 MEM_ADDR: 47 ACC: 108 COMMAND: out [1, 'str']
  DEBUG   isa:perform       out: "< What is your name?\n> Fedor\n< Hell" << "o"
  DEBUG   control_unit:start         PC: 118 TICK: 247 P_ADDR: 11 MEM_ADDR: 48 ACC: 111 COMMAND: jmp [9]
  DEBUG   control_unit:start         PC: 119 TICK: 248 P_ADDR: 9 MEM_ADDR: 48 ACC: 111 COMMAND: jz [12]
  DEBUG   control_unit:start         PC: 120 TICK: 249 P_ADDR: 10 MEM_ADDR: 48 ACC: 111 COMMAND: out [1, 'str']
  DEBUG   isa:perform       out: "< What is your name?\n> Fedor\n< Hello" << ","
  DEBUG   control_unit:start         PC: 121 TICK: 253 P_ADDR: 11 MEM_ADDR: 49 ACC: 44 COMMAND: jmp [9]
  DEBUG   control_unit:start         PC: 122 TICK: 254 P_ADDR: 9 MEM_ADDR: 49 ACC: 44 COMMAND: jz [12]
  DEBUG   control_unit:start         PC: 123 TICK: 255 P_ADDR: 10 MEM_ADDR: 49 ACC: 44 COMMAND: out [1, 'str']
  DEBUG   isa:perform       out: "< What is your name?\n> Fedor\n< Hello," << " "
  DEBUG   control_unit:start         PC: 124 TICK: 259 P_ADDR: 11 MEM_ADDR: 50 ACC: 32 COMMAND: jmp [9]
  DEBUG   control_unit:start         PC: 125 TICK: 260 P_ADDR: 9 MEM_ADDR: 50 ACC: 32 COMMAND: jz [12]
  DEBUG   control_unit:start         PC: 126 TICK: 261 P_ADDR: 10 MEM_ADDR: 50 ACC: 32 COMMAND: out [1, 'str']
  DEBUG   control_unit:start         PC: 127 TICK: 263 P_ADDR: 11 MEM_ADDR: 50 ACC: 0 COMMAND: jmp [9]
  DEBUG   control_unit:start         PC: 128 TICK: 264 P_ADDR: 9 MEM_ADDR: 50 ACC: 0 COMMAND: jz [12]
  DEBUG   control_unit:start         PC: 129 TICK: 266 P_ADDR: 12 MEM_ADDR: 50 ACC: 0 COMMAND: mov [55]
  DEBUG   control_unit:start         PC: 130 TICK: 268 P_ADDR: 13 MEM_ADDR: 55 ACC: 0 COMMAND: jz [16]
  DEBUG   control_unit:start         PC: 131 TICK: 269 P_ADDR: 14 MEM_ADDR: 55 ACC: 0 COMMAND: out [1, 'str']
  DEBUG   isa:perform       out: "< What is your name?\n> Fedor\n< Hello, " << "F"
  DEBUG   control_unit:start         PC: 132 TICK: 273 P_ADDR: 15 MEM_ADDR: 56 ACC: 70 COMMAND: jmp [13]
  DEBUG   control_unit:start         PC: 133 TICK: 274 P_ADDR: 13 MEM_ADDR: 56 ACC: 70 COMMAND: jz [16]
  DEBUG   control_unit:start         PC: 134 TICK: 275 P_ADDR: 14 MEM_ADDR: 56 ACC: 70 COMMAND: out [1, 'str']
  DEBUG   isa:perform       out: "< What is your name?\n> Fedor\n< Hello, F" << "e"
  DEBUG   control_unit:start         PC: 135 TICK: 279 P_ADDR: 15 MEM_ADDR: 57 ACC: 101 COMMAND: jmp [13]
  DEBUG   control_unit:start         PC: 136 TICK: 280 P_ADDR: 13 MEM_ADDR: 57 ACC: 101 COMMAND: jz [16]
  DEBUG   control_unit:start         PC: 137 TICK: 281 P_ADDR: 14 MEM_ADDR: 57 ACC: 101 COMMAND: out [1, 'str']
  DEBUG   isa:perform       out: "< What is your name?\n> Fedor\n< Hello, Fe" << "d"
  DEBUG   control_unit:start         PC: 138 TICK: 285 P_ADDR: 15 MEM_ADDR: 58 ACC: 100 COMMAND: jmp [13]
  DEBUG   control_unit:start         PC: 139 TICK: 286 P_ADDR: 13 MEM_ADDR: 58 ACC: 100 COMMAND: jz [16]
  DEBUG   control_unit:start         PC: 140 TICK: 287 P_ADDR: 14 MEM_ADDR: 58 ACC: 100 COMMAND: out [1, 'str']
  DEBUG   isa:perform       out: "< What is your name?\n> Fedor\n< Hello, Fed" << "o"
  DEBUG   control_unit:start         PC: 141 TICK: 291 P_ADDR: 15 MEM_ADDR: 59 ACC: 111 COMMAND: jmp [13]
  DEBUG   control_unit:start         PC: 142 TICK: 292 P_ADDR: 13 MEM_ADDR: 59 ACC: 111 COMMAND: jz [16]
  DEBUG   control_unit:start         PC: 143 TICK: 293 P_ADDR: 14 MEM_ADDR: 59 ACC: 111 COMMAND: out [1, 'str']
  DEBUG   isa:perform       out: "< What is your name?\n> Fedor\n< Hello, Fedo" << "r"
  DEBUG   control_unit:start         PC: 144 TICK: 297 P_ADDR: 15 MEM_ADDR: 60 ACC: 114 COMMAND: jmp [13]
  DEBUG   control_unit:start         PC: 145 TICK: 298 P_ADDR: 13 MEM_ADDR: 60 ACC: 114 COMMAND: jz [16]
  DEBUG   control_unit:start         PC: 146 TICK: 299 P_ADDR: 14 MEM_ADDR: 60 ACC: 114 COMMAND: out [1, 'str']
  DEBUG   isa:perform       out: "< What is your name?\n> Fedor\n< Hello, Fedor" << "\n"
  DEBUG   control_unit:start         PC: 147 TICK: 303 P_ADDR: 15 MEM_ADDR: 61 ACC: 10 COMMAND: jmp [13]
  DEBUG   control_unit:start         PC: 148 TICK: 304 P_ADDR: 13 MEM_ADDR: 61 ACC: 10 COMMAND: jz [16]
  DEBUG   control_unit:start         PC: 149 TICK: 305 P_ADDR: 14 MEM_ADDR: 61 ACC: 10 COMMAND: out [1, 'str']
  DEBUG   control_unit:start         PC: 150 TICK: 307 P_ADDR: 15 MEM_ADDR: 61 ACC: 0 COMMAND: jmp [13]
  DEBUG   control_unit:start         PC: 151 TICK: 308 P_ADDR: 13 MEM_ADDR: 61 ACC: 0 COMMAND: jz [16]
  DEBUG   control_unit:start         PC: 152 TICK: 310 P_ADDR: 16 MEM_ADDR: 61 ACC: 0 COMMAND: halt []
